### DISCUSSION TITLE
Does DiskCache work with Netlify functions (AWS lambdas)?

### DISCUSSION THREAD

**Discussion by arpitdalal** [Josh-McFarlin/remix-image#23] [Q&A] [answered]
Docs say to install `@next-boost/hybrid-disk-cache` pkg but is that only for Vercel?
AWS Lambdas does support `temp` storage up to 512MB I believe, should I be able to cache my images there with this lib?

**Comment by Josh-McFarlin** [comment, marked as answer]
That's a great question. Since Lambda supports `tmp`, DiskCache should work. If you try it and it does not work, you may need to change the DiskCache dir, you can see this in [the Vercel example](https://github.com/Josh-McFarlin/remix-image/blob/master/examples/vercel/app/routes/api/image.ts).

However, `tmp` only holds up to 512 MB and it will eventually be reset between invocations. If you want better performance (cache more transformations, cache transformations longer, faster loading), a better option would be Lambda EFS. To use EFS, you would need to [create your own Cache](https://remix-image.mcfarl.in/docs/advanced/creating-a-cache) and provide it to the loader. If this is a small project / performance isn't critical, `tmp` will work fine and it's free. 

**Edit**: Actually, I have never used EFS in Lambda before, so I was not sure how it worked. After reading up on it, you likely do not need a custom Cache. You should be able to create EFS through AWS, mount it to your Lambda, and use DiskCache, setting the `path` property to your mount location. 

And yes, you would need to install that package. Unfortunately it cannot be included in Remix-Image, it must be installed separately. 

The config for the EFS solution would be something along the lines of this:

```typescript
const prodAwsUrl = process.env.DEPLOYMENT_URL;
const prodAwsEFS = process.env.EFS_DIR;

const config = {
  selfUrl: process.env.NODE_ENV === "development" ? "http://localhost:3000" : prodAwsUrl,
  cache: new DiskCache({
    path: process.env.NODE_ENV === "development" ? path.join(os.tmpdir(), "img") : prodAwsEFS,
  }),
  resolver: fetchImage,
  transformer: sharpTransformer,
  basePath: process.env.NODE_ENV === "development" ? "public" : "/",
};
```

**Reply by arpitdalal** [reply]
This sounds about right, I am not planning on going down the EFS road just yet but appreciate your quick little example which can be super helpful. I've a question regarding the `selfUrl`, how important is that to get it right? I can't find where to get that url in prod.

**Reply by Josh-McFarlin** [reply]
A good way to do `selfUrl` is an environment variable that is set to be the URL of your project once it is deployed, something like `https://example.com`

**Reply by arpitdalal** [reply]
> A good way to do `selfUrl` is an environment variable that is set to be the URL of your project once it is deployed, something like `https://example.com`

Even if it is a server-less environment?
